packages/arb-token-bridge-ui/src/hooks/useRedeemRetryable.ts

```diff
@@ -1,92 +1,92 @@
 
-export function useRedeemRetryable(): UseRedeemRetryableResult {
-  const {
-    app: { arbTokenBridge }
-  } = useAppState()
-  const [networks] = useNetworks()
-  const { childChain, parentChainProvider } = useNetworksRelationship(networks)
-  const { data: signer } = useSigner()
-  const l2NetworkName = getNetworkName(childChain.id)
+export function useRedeemRetryable(
+  tx: MergedTransaction,
+  address: `0x${string}` | undefined
+): UseRedeemRetryableResult {
+  const { data: signer } = useSigner({ chainId: tx.childChainId })
+  const { updatePendingTransaction } = useTransactionHistory(address)
+
+  const l2NetworkName = getNetworkName(tx.childChainId)
 
   const [isRedeeming, setIsRedeeming] = useState(false)
 
-  async function redeem(tx: MergedTransaction) {
+  const redeem = useCallback(async () => {
     if (isRedeeming) {
       return
     }
-
-    let retryableTicket: IL1ToL2MessageWriter
-
     try {
       setIsRedeeming(true)
 
       if (!signer) {
         throw 'Signer is undefined'
       }
 
-      retryableTicket = await getRetryableTicket({
+      const retryableTicket = await getRetryableTicket({
         l1TxHash: tx.txId,
         retryableCreationId: tx.l1ToL2MsgData?.retryableCreationTxID,
-        l1Provider: parentChainProvider,
+        l1Provider: getProviderForChainId(tx.parentChainId),
         l2Signer: signer
       })
-    } catch (error: any) {
-      setIsRedeeming(false)
-      return errorToast(
-        `There was an error, here is more information: ${error.message}`
-      )
-    }
 
-    try {
-      const tx = await retryableTicket.redeem()
-      await tx.wait()
-    } catch (error: any) {
-      if (isUserRejectedError(error)) {
-        return
+      const reedemTx = await retryableTicket.redeem()
+      await reedemTx.wait()
+
+      const status = await retryableTicket.status()
+      const isSuccess = status === L1ToL2MessageStatus.REDEEMED
+
+      const redeemReceipt = (await retryableTicket.getSuccessfulRedeem()) as {
+        status: L1ToL2MessageStatus.REDEEMED
+        l2TxReceipt: TransactionReceipt
       }
 
-      return errorToast(
-        `There was an error, here is more information: ${error.message}`
-      )
-    } finally {
+      updatePendingTransaction({
+        ...tx,
+        l1ToL2MsgData: {
+          l2TxID: redeemReceipt.l2TxReceipt.transactionHash,
+          status,
+          retryableCreationTxID: retryableTicket.retryableCreationId,
+          fetchingUpdate: false
+        },
+        resolvedAt: isSuccess ? dayjs().valueOf() : null,
+        depositStatus: isSuccess ? DepositStatus.L2_SUCCESS : tx.depositStatus
+      })
+
       setIsRedeeming(false)
 
       // track in analytics
       if (shouldTrackAnalytics(l2NetworkName)) {
         trackEvent('Redeem Retryable', { network: l2NetworkName })
       }
-    }
+    } catch (error: any) {
+      setIsRedeeming(false)
 
-    // update in store
-    arbTokenBridge.transactions.fetchAndUpdateL1ToL2MsgStatus(
```

Comment:
since we stopped calling `fetchAndUpdateL1ToL2MsgStatus` fro `arbTokenBridge.transactions.`, can you please delete it from `arbTokenBridge.transactions.`?